### PR TITLE
Support to clear schema before loading

### LINKED ISSUE #134: Add load option used to delete schema before loading

### PR DESCRIPTION
**Pull request by Linary**
Fix #134

Change-Id: Icc8c096e6c1539370ccfcd83b4a9753ab03040b1

### CODE REVIEW

File: src/test/java/com/baidu/hugegraph/loader/test/functional/FileLoadTest.java [hunk 1]
```diff
@@ -215,6 +219,60 @@ public void testNoSchemaFile() {
...
+    public void testClearSchemaBeforeLoad() {
+        LoadOptions options = new LoadOptions();
+        options.host = Constants.HTTP_PREFIX + SERVER;
+        options.port = PORT;
+        options.graph = GRAPH;
+        HugeClient client = HugeClientHolder.get(options);
+        SchemaManager schema = client.schema();
+        schema.propertyKey("name").asText().ifNotExist().create();
+        schema.propertyKey("age").asInt().ifNotExist().create();
+        // The datatype is int
```
Comment (zhoney) [inline src/test/java/com/baidu/hugegraph/loader/test/functional/FileLoadTest.java]: The old datatype of city is int

File: src/main/java/com/baidu/hugegraph/loader/HugeGraphLoader.java [hunk 2]
```diff
@@ -99,6 +102,40 @@ public void load() {
         this.stopThenShutdown();
     }
 
+    private void clearSchemaIfNeeded() {
+        LoadOptions options = this.context.options();
+        if (!options.clearOldSchema) {
+            return;
+        }
+
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/loader/HugeGraphLoader.java]: add LOG.info() for clear schema

File: src/main/java/com/baidu/hugegraph/loader/HugeGraphLoader.java [hunk 3]
```diff
@@ -99,6 +102,40 @@ public void load() {
...
+        HugeClient client = HugeClientHolder.get(options);
+        SchemaManager schema = client.schema();
+        com.baidu.hugegraph.driver.TaskManager task = client.task();
+        int timeout = options.clearSchemaTimeout;
+        // Clear schema
+        List<Long> taskIds = new ArrayList<>();
+        schema.getIndexLabels().forEach(il -> {
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/loader/HugeGraphLoader.java]: why not use clear api(more efficient)?

File: src/test/resources/clear_schema_before_load/struct_hdfs.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "vertices": [
+    {
+      "label": "person",
+      "input": {
+        "type": "${source_type}",
+        "path": "${store_path}/vertex_person.csv",
+        "core_site_path": "src/test/resources/hdfs_with_core_site_path/core-site.xml",
```
Comment (javeme) [inline src/test/resources/clear_schema_before_load/struct_hdfs.json]: unused core_site_path?

File: src/main/java/com/baidu/hugegraph/loader/HugeGraphLoader.java (hunk 3 above)
Comment (Linary) [reply src/main/java/com/baidu/hugegraph/loader/HugeGraphLoader.java]: LGTM!

File: src/test/resources/clear_schema_before_load/struct_hdfs.json (hunk 4 above)
Comment (Linary) [reply src/test/resources/clear_schema_before_load/struct_hdfs.json]: No, hdfs need it connect to namenode

File: src/main/java/com/baidu/hugegraph/loader/executor/LoadOptions.java [hunk 5]
```diff
@@ -57,6 +57,10 @@
                description = "The token of graph for authentication")
     public String token = null;
 
+    @Parameter(names = {"--clear-old-schema"}, arity = 1,
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/loader/executor/LoadOptions.java]: prefer clear-all

File: src/main/java/com/baidu/hugegraph/loader/HugeGraphLoader.java [hunk 6]
```diff
@@ -99,6 +101,19 @@ public void load() {
         this.stopThenShutdown();
     }
 
+    private void clearSchemaIfNeeded() {
+        LoadOptions options = this.context.options();
+        if (!options.clearOldSchema) {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/loader/HugeGraphLoader.java]: clearOldData

File: src/main/java/com/baidu/hugegraph/loader/executor/LoadOptions.java [hunk 7]
```diff
@@ -57,6 +57,10 @@
                description = "The token of graph for authentication")
     public String token = null;
 
+    @Parameter(names = {"--clear-all"}, arity = 1,
+               description = "Whether to clear old schema before loading")
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/loader/executor/LoadOptions.java]: clear old data

File: src/main/java/com/baidu/hugegraph/loader/executor/LoadOptions.java [hunk 8]
```diff
@@ -57,6 +57,10 @@
                description = "The token of graph for authentication")
     public String token = null;
 
+    @Parameter(names = {"--clear-all"}, arity = 1,
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/loader/executor/LoadOptions.java]: clear-all-data

File: src/main/java/com/baidu/hugegraph/loader/executor/LoadOptions.java [hunk 9]
```diff
@@ -57,6 +57,10 @@
...
+               description = "Whether to clear old schema before loading")
+    public boolean clearOldSchema = false;
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/loader/executor/LoadOptions.java]: ditto

File: src/main/java/com/baidu/hugegraph/loader/HugeGraphLoader.java [hunk 10]
```diff
@@ -81,6 +81,8 @@ private void addShutdownHook() {
 
     public void load() {
         try {
+            // Clear schema if needed
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/loader/HugeGraphLoader.java]: keep clearTimeout arg and set default 240s(cassandra cluster needs 180+s)

**Pull request merged**